editor/src/core/shared/template-path.ts

```diff
@@ -234,139 +118,40 @@ export function templatePath(fullElementPath: ElementPath[]): TemplatePath {
+    pathCache.cached = newTemplatePath(fullElementPath)
   }
 
   return pathCache.cached
 }
 
-function scenePath(elementPaths: ElementPath[]): ScenePath {
-  if (isEmptyElementPathsArray(elementPaths)) {
-    return emptyScenePath
-  }
-
-  const pathCache = getScenePathCache(elementPaths)
-  if (pathCache.cached == null) {
-    const newPath = newScenePath(elementPaths)
-    pathCache.cached = newPath
-    return newPath
-  } else {
-    return pathCache.cached
-  }
-}
-
-export function staticScenePath(elementPaths: StaticElementPath[]): StaticScenePath {
-  return scenePath(elementPaths) as StaticScenePath
-}
-
-function instancePath(scene: ScenePath, elementPath: ElementPath): InstancePath {
-  if (scene.sceneElementPaths.length === 0 && elementPath.length === 0) {
-    return emptyInstancePath
-  } else {
-    const pathCache = getInstancePathCache(scene, elementPath)
-    if (pathCache.cached == null) {
-      const newPath = newInstancePath(scene, elementPath)
-      pathCache.cached = newPath
-      return newPath
-    } else {
-      return pathCache.cached
-    }
-  }
-}
-
-export function staticInstancePath(
-  scene: StaticScenePath,
-  elementPath: StaticElementPath,
-): StaticInstancePath {
-  return instancePath(scene, elementPath) as StaticInstancePath
-}
-
 export function asStatic(path: TemplatePath): StaticTemplatePath {
   return path as StaticTemplatePath
 }
 
-export function isScenePath(path: unknown): path is ScenePath {
-  return (path as any)?.type === 'scenepath'
-}
-
-export function isInstancePath(path: unknown): path is InstancePath {
-  return (path as any)?.scene != null && (path as any)?.element != null
-}
-
 export function isTemplatePath(path: unknown): path is TemplatePath {
-  return isScenePath(path) || isInstancePath(path)
+  return (path as any)?.type === 'templatepath'
 }
 
-export function isTopLevelInstancePath(path: TemplatePath): path is InstancePath {
-  return isInstancePath(path) && path.element.length === 1
+export function isRootElementOfInstance(path: TemplatePath): boolean {
+  return path.parts.length > 1 && last(path.parts)!.length === 1
 }
 
 export function isStoryboardPath(path: TemplatePath): boolean {
-  const fullElementPath = fullElementPathForPath(path)
-  return fullElementPath.length === 1 && fullElementPath[0].length === 1
+  return path.parts.length === 1 && path.parts[0].length === 1
 }
 
 export function isStoryboardDescendant(path: TemplatePath): boolean {
-  const asInstancePath = instancePathForElementAtPathDontThrowOnScene(path)
-  return isEmptyScenePath(asInstancePath.scene) && !isStoryboardPath(asInstancePath)
+  return path.parts.length === 1 && path.parts[0].length > 1
 }
 
 export function isStoryboardChild(path: TemplatePath): boolean {
-  const asInstancePath = instancePathForElementAtPathDontThrowOnScene(path)
-  return isEmptyScenePath(asInstancePath.scene) && asInstancePath.element.length === 2
+  return path.parts.length === 1 && path.parts[0].length === 1
```

Comment:
This is the same as `isStoryboardPath`, is that right?

Reply:
Oops!

Reply:
Fixed (and added tests covering it).